sdk/cosmos/azure-cosmos/docs/ErrorCodesAndRetries.md

```diff
@@ -0,0 +1,18 @@
+| 401   | For all operations: </br><ul><li> This is an unauthorized exception due to invalid auth tokens being used for the request. The client does NOT retry requests when this exception is encountered.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
+| 403    | <ul><li>For Substatus 3 (Write Forbidden): </br><ul><li>This exception occurs when a geo-replicated database account runs into writable/readable location changes (say, after a failover).</li><li>This exception can occur regardless of the Consistency level set for the account. </li><li>The client refreshes it's locational endpoints and retries requests when the user has enabled endpoint discovery in their client (default behavior).</li></ul></li><li>For all other cases: </br><ul><li> The client does NOT retry requests when this exception is encountered. </li>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```

Comment:
403/3 -> it probably only related to write region change, read will get a different error code I think

and in python, there is no handling for 403/1008 then?

Reply:
The client refreshes it's `locational endpoints` ->` location endpoints`?

Reply:
yep, like you said the current policy that takes care of 403's (endpointDiscoveryRetryPolicy) only takes care of 403/3 scenarios